qiskit_nature/drivers/second_quantization/electronic_structure_molecule_driver.py

```diff
@@ -0,0 +1,171 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+This module implements a common molecule electronic structure based driver.
+"""
+
+from typing import Optional, Dict, Any, Type
+import logging
+from enum import Enum
+
+from qiskit.exceptions import MissingOptionalLibraryError
+from .electronic_structure_driver import ElectronicStructureDriver, MethodType
+from ..molecule import Molecule
+from .pyscfd import PySCFDriver
+from .psi4d import PSI4Driver
+from .pyquanted import PyQuanteDriver
+from .gaussiand import GaussianDriver
+
+logger = logging.getLogger(__name__)
+
+
+class ElectronicStructureDriverType(Enum):
+    """Electronic structure Driver Type."""
+
+    AUTO = "auto"
+    PYSCF = "pyscf"
+    PSI4 = "psi4"
+    PYQUANTE = "pyquante"
+    GAUSSIAN = "gaussian"
```

Comment:
I am not sure whether that will work but rather than having the values be strings, could we not set the classes here directly?
Below you could then do something along the lines of:
```python
if driver_type == ElectronicStructureDriverType.AUTO:
    ...
else:
    driver_class = driver_type.value
    driver_class.check_installed()
```

Reply:
Thanks! I see now that I was simplifying it a bit too much because my proposal would have involved importing all drivers :sweat_smile: But the current approach looks good to me as it simplifies future addition of drivers :+1: